src/Integration.Vsix.UnitTests/Commands/ConnectedModeMenu/SaveSharedConnectionCommandTests.cs

```diff
@@ -0,0 +1,115 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Windows;
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.ConnectedMode.Shared;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.Vsix.Commands.ConnectedModeMenu;
+using SonarLint.VisualStudio.Integration.Vsix.Resources;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Commands.ConnectedModeMenu
+{
+    [TestClass]
+    public class SaveSharedConnectionCommandTests
+    {
+        [DataRow(SonarLintMode.Standalone, false)]
+        [DataRow(SonarLintMode.Connected, true)]
+        [DataRow(SonarLintMode.LegacyConnected, true)]
+        [TestMethod]
+        public void QueryStatus_EnableCommandCorrectly(SonarLintMode mode, bool expectedResult)
+        {
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+
+            var bindingConfiguration = new BindingConfiguration(null, mode, null);
+
+            var configurationProvider = CreateConfigurationProvider(bindingConfiguration);
+            SaveSharedConnectionCommand testSubject = CreateTestSubject(configurationProvider);
+
+            testSubject.QueryStatus(command, null);
+
+            command.Enabled.Should().Be(expectedResult);
+        }
+
+        [TestMethod]
+        public void Invoke_Success_InvokesCorrectly()
+        {
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+
+            var project = new BoundSonarQubeProject(new Uri("http://127.0.0.1:9000"), "projectKey", null);
+            var bindingConfiguration = new BindingConfiguration(project, SonarLintMode.Connected, null);
+
+            var sharedBindingConfigProvider = new Mock<ISharedBindingConfigProvider>();
+            sharedBindingConfigProvider.Setup(x => x.SaveSharedBinding(It.Is<SharedBindingConfigModel>(y => y.Uri == "http://127.0.0.1:9000/" && y.ProjectKey == "projectKey" && y.Organization == null))).Returns(true);
+
+            var messageBox = new Mock<IMessageBox>();
+
+            var configurationProvider = CreateConfigurationProvider(bindingConfiguration);
+            SaveSharedConnectionCommand testSubject = CreateTestSubject(configurationProvider, sharedBindingConfigProvider.Object, messageBox.Object);
+
+            testSubject.Invoke(command, null);
+
+            sharedBindingConfigProvider.Verify(x => x.SaveSharedBinding(It.Is<SharedBindingConfigModel>(y => y.Uri == "http://127.0.0.1:9000/" && y.ProjectKey == "projectKey" && y.Organization == null)), Times.Once);
```

Comment:
no tests with organization != null

Reply:
I don't think there's a need for that. It's just to see if the data is sent to the provider. Tests for those scenarios are there.